core/common/src/main/java/alluxio/collections/NonUniqueFieldIndex.java

```diff
@@ -51,8 +51,11 @@ public void add(T object) {
       synchronized (objSet) {
         // Adds the value to the object set.
         objSet.add(object);
+        if (objSet == mIndexMap.get(fieldValue)) {
```

Comment:
Can we do this the other way? Meaning, check to see if the `objSet` is the same, if not, `continue`, otherwise, just `add` and `break`. Then, we don't have to add objects to unnecessary sets.